### PR TITLE
Split gasman.c into gasman and HPC-GAP parts

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This is the hpcgap half of my gasman merge. In particular, with this patch gasman.h and gasman.c are identical in GAP and HPC-GAP.

### CODE REVIEW

**Comment by markuspf** [comment]
Shouldn't `hpc_boehm_gc.h` be in the directory `hpc/` (and it could then probably be called boehm_gc.h)?

**Comment by ChrisJefferson** [comment]
That would be sensible, now fixed (also synced with master)

File: src/hpc/boehm_gc.h [hunk 1]
```diff
@@ -0,0 +1,647 @@
...
+
+void CallbackForAllBags(
+     void (*func)(Bag) )
+{ }
+
+void StartRestoringBags( UInt nBags, UInt maxSize)
+{ }
+
+Bag NextBagRestoring( UInt size, UInt type)
+{ return 0; }
```
Comment (fingolfin) [inline src/hpc/boehm_gc.h]: Nitpick, but: this is a weird way to format a function. It also is different from what you do just a few lines ago.

Comment (fingolfin) [inline src/hpc/boehm_gc.h]: Also, the empty function bodies above are weird, but not as weird as that one :)

**Comment by fingolfin** [comment]
Thanks for the work!

Still, the formatting would be nice to get fixed soon (e.g. also the tabs instead of spaces...) :-)

**Comment by fingolfin** [comment]
(soon = before other changes are made, so that we don't obscure `git annotate` needlessly)

**Comment by ChrisJefferson** [comment]
I agree. The main reason to put this merge in "as is" is because (hopefully) it was just cutting+pasting the hpcgap / gap code. Now it is merged, we can clean it up.

**Pull request merged**